### ISSUE TITLE
CNNPatchPredictor predict method in wsi mode errors

### ISSUE THREAD

**Issue by R-J96** [TIA-Lab/tiatoolbox#130]
* TIA Toolbox version: 0.6.0
* Python version: 3.7.8
* Operating System: Ubuntu 20.04

### Description

I tested this both with the most recent develop and feature-patchwsi commits. When using the CNNPatchPredictor class to predict in wsi mode using the example arguments from the example notebook 05 the method errors and when using the updated arguments for the method not included in the notebook.

### What I Did
Running this as per the example in the notebook
```
wsi_output = predictor.predict([wsi_file_name],
                           mode='wsi',
                           stride_size=(112,112),
                           return_probabilities=True)
```

results in the error:
```
AttributeError                            Traceback (most recent call last)
<ipython-input-13-aeabd4a04e59> in <module>
      2                            mode='wsi',
      3                            stride_size=(112,112),
----> 4                            return_probabilities=True)

~/.conda/envs/tiatoolbox/lib/python3.7/site-packages/tiatoolbox-0.6.0-py3.7.egg/tiatoolbox/models/classification/patch_predictor.py in predict(self, img_list, mask_list, label_list, mode, return_probabilities, return_labels, on_gpu, patch_size, stride_size, resolution, units, merge_predictions, save_dir)
    539                     mask_path=img_mask,
    540                     patch_size=self.iostate.patch_size,
--> 541                     stride_size=self.iostate.stride_size,
    542                     resolution=self.iostate.input_resolutions[0]["resolution"],
    543                     units=self.iostate.input_resolutions[0]["units"],

AttributeError: '_IOStatePatchPredictor' object has no attribute 'stride_size'
```

Additionally, after looking through the source code and adding in the new arguments for the predict method:
```
wsi_output = predictor.predict([wsi_jp2],
                               mode='wsi',
                               patch_size=(244,244),
                               return_probabilities=True,
                               resolution=0.5,
                               units='mpp',
                               merge_predictions=True
                              )
```
this also errors with the following
```
AttributeError                            Traceback (most recent call last)
<ipython-input-31-270288848f10> in <module>
      5                                resolution=0.5,
      6                                units='mpp',
----> 7                                merge_predictions=True
      8                               )

~/.conda/envs/tiatoolbox/lib/python3.7/site-packages/tiatoolbox/models/classification/patch_predictor.py in predict(self, img_list, mask_list, label_list, mode, return_probabilities, return_labels, on_gpu, patch_size, stride_size, resolution, units, merge_predictions, save_dir)
    545                     stride_size=self.iostate.stride_size,
    546                     resolution=self.iostate.input_resolutions[0]['resolution'],
--> 547                     units=self.iostate.input_resolutions[0]['units'],
    548                 )
    549                 output_model = self._predict_engine(

~/.conda/envs/tiatoolbox/lib/python3.7/site-packages/tiatoolbox/models/dataset/classification.py in __init__(self, img_path, mode, mask_path, preproc_func, patch_size, stride_size, resolution, units, auto_get_mask)
    298             mask_reader = self.reader.tissue_mask(resolution=1.25, units="power")
    299             # ? will this mess up ?
--> 300             mask_reader.attach_to_reader(self.reader.info)
    301 
    302         if mask_reader is not None:

AttributeError: 'VirtualWSIReader' object has no attribute 'attach_to_reader'
```

Seems the answer to the comment in the source code is that yes this will mess up :).

**Comment by shaneahmed** [comment]
Please can you specify the image type? Is this WSI or a visual field?

**Comment by shaneahmed** [comment]
Fixed in #109
